var/spack/repos/builtin/packages/py-pyprecice/package.py

```diff
@@ -33,6 +35,7 @@ class PyPyprecice(PythonPackage):
     version("2.0.0.1", sha256="96eafdf421ec61ad6fcf0ab1d3cf210831a815272984c470b2aea57d4d0c9e0e")
 
     depends_on("precice@develop", when="@develop")
+    depends_on("precice@2.5.0", when="@2.5.0.1:2.5.0")
```

Comment:
These `when` args look odd. As is, this says that `py-precice@2.5.0.0` does not depend on `precice` at all. I think a better way of organizing these would be something like:
```python
depends_on("precice@2.5.0", when="@2.5.0")
```
Here, `@2.5.0` implies `@2.5.0.0:2.5.0.999`. If you want to get really fancy, you could do this in a for loop:
```python
for ver in ["develop", "2.5.0", "2.4.0", ..., "2.0.0"]:
    depends_on("precice@" + ver, when="@" + ver)
```

Reply:
Thanks for this, I made the change. However I am a bit confused about this now. If the `when` args were a problem for `2.5.0.0`, then how is it fine in the case of:
```python
depends_on("precice@2.4.0", when="@2.4.0.1:2.4.0")
```
is it because the version `2.4.0.1` does not exist and hence `2.4.0.0` is covered? I read up on the `when` args syntax in detail, but I guess I am still missing something.

Reply:
> If the when args were a problem for 2.5.0.0, then how is it fine in the case of:

2.4.0.0 is also broken.

> It would be awesome if the following was possible:

#14002 was actually introduced for this very reason, although it got stuck.